File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.py
def is_sorted(arr):
    low , high = 0 , 1
    while(high< len(arr)):
        if(arr[low] > arr[high]):
            return (False , high)
        low +=1
        high +=1

    return (True , None)



class Solution:
    def check(self, nums: List[int]) -> bool:    
        flag , val = is_sorted(nums)
        if(flag==True):
            return (True)
        else:
            first_half  , last_half =nums[:val] , nums[val:]
            last_half.extend(first_half)
            res , val = is_sorted(last_half)
            if(res == True):
                return (True)
            else:
                return(False)

File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.cs
public class Solution {
    public bool Check(int[] nums) {
        int low = 0;
			int high = 1;
			while (high < nums.Length)
			{
				if (nums[low] > nums[high])
				{
					break;
				}
                low++;
                high++;
			}

			if (high == nums.Length)
			{
				return(true);
			}


			List<int> checker = new List<int>();

			for (int i = high; i < nums.Length; i++)
			{
				checker.Add(nums[i]);
			}

			for (int i = 0; i < low + 1; i++)
			{
				checker.Add(nums[i]);
			}

			bool flag = false;
			for (int i = 1; i < checker.Count; i++)
			{
				if (checker[i - 1] > checker[i])
				{
					return (false);
				}
				else
				{
					flag = true;
				}
			}
			
			return (flag);
			
			


        
    }
}